astropy/time/tests/test_custom_formats.py

```diff
@@ -89,14 +92,14 @@ def value(self):
             return self.jd1, self.jd2
 
     t = Time.now()
-    with pytest.raises(AttributeError):
+    with pytest.raises(TypeError):
```

Comment:
Why these changes?

Reply:
In this case the raising of AttributeError was a result of the existing implementation and came with an impenetrable message and backtrace. The new code is able to detect the problem and raise whichever exception seems suitable; in this case TypeError seemed appropriate. Only mis-implemented non-functional custom formats - like those I worked through trying to get one working - trigger this exception, so I do not expect to break any code by changing the exception; the previous exception was never chosen, just an artifact of the previous implementation.